dbt/adapters/bigquery/python_submissions.py

```diff
@@ -1,187 +1,163 @@
-        )
-        self.job_client = self._get_job_client()
+        self._storage_client = storage_client(credentials)
+        self._project = credentials.execution_project
+        self._region = credentials.dataproc_region
 
-    def _upload_to_gcs(self, filename: str, compiled_code: str) -> None:
-        bucket = self.storage_client.get_bucket(self.credential.gcs_bucket)
-        blob = bucket.blob(filename)
-        blob.upload_from_string(compiled_code)
+        schema = parsed_model["schema"]
+        identifier = parsed_model["alias"]
+        self._model_file_name = f"{schema}/{identifier}.py"
+        self._gcs_bucket = credentials.gcs_bucket
+        self._gcs_path = f"gs://{credentials.gcs_bucket}/{self._model_file_name}"
 
-    def submit(self, compiled_code: str) -> dataproc_v1.types.jobs.Job:
-        # upload python file to GCS
-        self._upload_to_gcs(self.model_file_name, compiled_code)
-        # submit dataproc job
-        return self._submit_dataproc_job()
+        # set retry policy, default to timeout after 24 hours
+        retry = RetryFactory(credentials)
+        self._polling_retry = retry.polling(timeout=parsed_model["config"].get("timeout"))
 
-    def _get_job_client(
-        self,
-    ) -> Union[dataproc_v1.JobControllerClient, dataproc_v1.BatchControllerClient]:
-        raise NotImplementedError("_get_job_client not implemented")
+    def _upload_to_gcs(self, compiled_code: str) -> None:
+        bucket = self._storage_client.get_bucket(self._gcs_bucket)
+        blob = bucket.blob(self._model_file_name)
+        blob.upload_from_string(compiled_code)
 
-    def _submit_dataproc_job(self) -> dataproc_v1.types.jobs.Job:
-        raise NotImplementedError("_submit_dataproc_job not implemented")
 
+class ClusterDataprocHelper(_BaseDataProcHelper):
+    def __init__(self, parsed_model: Dict, credentials: BigQueryCredentials) -> None:
+        super().__init__(parsed_model, credentials)
+        self._job_controller_client = job_controller_client(credentials)
+        self._cluster_name = parsed_model["config"].get(
+            "dataproc_cluster_name", credentials.dataproc_cluster_name
+        )
 
-class ClusterDataprocHelper(BaseDataProcHelper):
-    def _get_job_client(self) -> dataproc_v1.JobControllerClient:
-        if not self._get_cluster_name():
+        if not self._cluster_name:
             raise ValueError(
                 "Need to supply dataproc_cluster_name in profile or config to submit python job with cluster submission method"
             )
-        return dataproc_v1.JobControllerClient(
-            client_options=self.client_options, credentials=self.GoogleCredentials
-        )
 
-    def _get_cluster_name(self) -> str:
-        return self.parsed_model["config"].get(
-            "dataproc_cluster_name", self.credential.dataproc_cluster_name
-        )
+    def submit(self, compiled_code: str) -> Job:
+        _logger.info(f"Submitting cluster job to: {self._cluster_name}")
```

Comment:
nit: These can probably be `debug`, the user will know the cluster they are submitting to / will see it if it fails

Reply:
I'm fine with that. I'm also fine just removing it. The only reason it's here is because that's what the current process is doing.